### PR TITLE
fix: fix table existence validation function

### PR DESCRIPTION
**Pull request by bkyryliuk**
### SUMMARY
* fixes table validation in API
* drops incorrect unique constraint for the mysql db [ probably a miss ]

### TEST PLAN
* unit tests

### CODE REVIEW

File: superset/connectors/sqla/models.py [hunk 1]
```diff
@@ -453,7 +453,7 @@ class SqlaTable(  # pylint: disable=too-many-public-methods,too-many-instance-at
     owner_class = security_manager.user_model
 
     __tablename__ = "tables"
-    __table_args__ = (UniqueConstraint("database_id", "table_name"),)
+    __table_args__ = (UniqueConstraint("database_id", "schema", "table_name"),)
```
Comment (bkyryliuk) [inline superset/connectors/sqla/models.py:456]: @john-bodley - do you know if there are any issues with changing __table_args__ here?

Comment (john-bodley) [reply superset/connectors/sqla/models.py:456]: This will not work per https://github.com/apache/incubator-superset/pull/5449.

Comment (bkyryliuk) [reply superset/connectors/sqla/models.py]: thanks @john-bodley, reverted this line

**Review by nytai — APPROVED** [review]
This looks like a good enough solution, given the constraints.

**Pull request merged**